src/core/include/mp-units/cartesian_vector.h

```diff
@@ -267,28 +288,49 @@ class cartesian_vector : public detail::cartesian_vector_iface {
   }
 
 #if MP_UNITS_HOSTED
-  friend constexpr std::ostream& operator<<(std::ostream& os, const cartesian_vector& vec)
+  friend std::ostream& operator<<(std::ostream& os, const cartesian_vector& vec)
   {
     return os << '[' << vec[0] << ", " << vec[1] << ", " << vec[2] << ']';
   }
 #endif
 };
 
 template<typename Arg, typename... Args>
-  requires(sizeof...(Args) <= 2) && requires { typename std::common_type_t<Arg, Args...>; }
+  requires(sizeof...(Args) == 2) && requires { typename std::common_type_t<Arg, Args...>; }
 cartesian_vector(Arg, Args...) -> cartesian_vector<std::common_type_t<Arg, Args...>>;
 
+template<typename T, typename U>
+[[nodiscard]] constexpr auto dot(const cartesian_vector<T>& lhs, const cartesian_vector<U>& rhs)
+{
+  return scalar_product(lhs, rhs);
+}
+
+template<typename T, typename U>
+[[nodiscard]] constexpr auto cross(const cartesian_vector<T>& lhs, const cartesian_vector<U>& rhs)
+{
+  return vector_product(lhs, rhs);
+}
+
+template<typename T>
+  requires treat_as_floating_point<T>
+[[nodiscard]] constexpr auto norm(const cartesian_vector<T>& vec)
+{
+  return magnitude(vec);
+}
+
+template<detail::Scalar T>
+inline constexpr bool is_vector<cartesian_vector<T>> = true;
+
 }  // namespace mp_units
 
 #if MP_UNITS_HOSTED
-// TODO use parse and use formatter for the underlying type
 template<typename T, typename Char>
 struct MP_UNITS_STD_FMT::formatter<mp_units::cartesian_vector<T>, Char> :
     formatter<std::basic_string_view<Char>, Char> {
-  template<typename FormatContext>
-  auto format(const mp_units::cartesian_vector<T>& vec, FormatContext& ctx) const
+  template<typename Ctx>
+  auto format(const mp_units::cartesian_vector<T>& v, Ctx& ctx) const
   {
-    return format_to(ctx.out(), "[{}, {}, {}]", vec[0], vec[1], vec[2]);
+    return format_to(ctx.out(), "[{}, {}, {}]", v[0], v[1], v[2]);
```

Comment:
Please restore the previous code. `FormatContext` provides more info and is used consistently in the entire project for all formatters.

`v` is too short as an identifier and often causes some stupid shadowing errors in MSVC.